routing_common/transit_header.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include <cstdint>
+
+namespace routing
+{
+namespace transit
+{
+struct TransitHeader final
+{
+  TransitHeader() { Reset(); }
+  TransitHeader(uint16_t version, uint32_t gatesOffset, uint32_t edgesOffset,
+                uint32_t transfersOffset, uint32_t linesOffset, uint32_t shapesOffset,
+                uint32_t networksOffset, uint32_t endOffset);
+  void Reset();
+  bool IsEqualForTesting(TransitHeader const & header) const;
+
+  template <class TSink>
+  void Serialize(TSink & sink) const
+  {
+    WriteToSink(sink, m_version);
```

Comment:
I strongly suggest you to get rid of these explicit Serialize and Deserialize, and use visitors approach instead, as it allows you to easily implement serdes to binary, json, debug prints etc. Take a look at the ugc/types.hpp

Reply:
Yes, thank you. Done.